.github/workflows/ubuntu_clean_meson_build.yml

```diff
@@ -49,7 +46,7 @@ jobs:
           -Dcapi-ml-inference-actual=capi-ml-inference \
           -Denable-capi=enabled \
           ${{ matrix.meson_options }} \
-          build
-    - run: ninja -C build
+          build-${{matrix.meson_options}}-${{matrix.os}}
```

Comment:
After removal of ```meson setup build```, you won't see the build directory. Do you really need this?

(from meson manual, ```meson setup build``` == ```meson build```):
```
The setup command
       Using Meson is simple and follows the common two-phase process of most build systems. First you run Meson to configure your build:

       meson setup [ options ] [ build directory ] [ source directory ]

       Note that the build directory must be different from the source directory. Meson does not support building inside the source directory and attempting to do that leads to an error.

       After a successful configuration step you can build the source by running the actual build command in the build directory. The default backend of Meson is Ninja, which can be invoked like this.

       ninja [ target ]

       You only need to run the Meson command once: when you first configure your build dir. After that you just run the build command. Meson will autodetect changes in your source tree and regenerate all files needed to build  the
       project.

       The setup command is the default operation. If no actual command is specified, Meson will assume you meant to do a setup. That means that you can set up a build directory without the setup command like this:

       meson [ options ] [ build directory ] [ source directory ]

```

Reply:
When running matrix jobs in GitHub Actions, they run concurrently,so I changed to using the format "build-~~" instead of just "build" as the directory name to avoid potential issues with executing subsequent commands if there were directories with the same name.

Reply:
They are running separated.

When renaming is dropped, you may consider it's approved.